script/playbackstrategy/msestrategy.js

```diff
@@ -120,6 +141,57 @@ define('bigscreenplayer/playbackstrategy/msestrategy',
         }
       }
 
+      function createPlaybackProperties () {
+        var playbackProperties = {};
+
+        playbackProperties.seekable_range = getSeekableRange().start + ' to ' + getSeekableRange().end;
+        playbackProperties.current_time = getCurrentTime();
+        playbackProperties.duration = getDuration();
+
+        return playbackProperties;
+      }
+
+      function propagateCdnFailover (event, cdn) {
+        // Initial playback
+        if (cdn === mediaSources[0].cdn) return;
```

Comment:
Should we get into this code on initial playback? It looks like we only get here when dashjs throws a CDN_FAILOVER event.

Reply:
onCdnFailover gets called every time there is a BASEURL_SELECTED event fired from dashjs. I would have preferred to use SERVICE_LOCATION_BLACKLIST_ADD or SERVICE_LOCATION_BLACKLIST_CHANGED but then we don't get the baseurl back to match up with our other information